===========================================================
 Ignore touchpads when swapping buttons

  We always want clicks on the touchpad
to be left click (bug 324721)

diff --git a/plugins/mouse/gsd-mouse-manager.c b/plugins/mouse/gsd-mouse-manager.c
--- a/plugins/mouse/gsd-mouse-manager.c
+++ b/plugins/mouse/gsd-mouse-manager.c
@@ -233,6 +233,11 @@ xinput_device_has_buttons (XDeviceInfo *device_info)
         int i;
         XAnyClassInfo *class_info;
 
+        if (device_info->type == gdk_x11_get_xatom_by_name (XI_TABLET) ||
+            device_info->type == gdk_x11_get_xatom_by_name (XI_TOUCHSCREEN) ||
+            device_info->type == gdk_x11_get_xatom_by_name (XI_TOUCHPAD))
+                return FALSE;
+
         class_info = device_info->inputclassinfo;
         for (i = 0; i < device_info->num_classes; i++) {
                 if (class_info->class == ButtonClass) {
@@ -356,6 +361,7 @@
 #ifdef HAVE_X11_EXTENSIONS_XINPUT_H
         if (supports_xinput_devices ()) {
                 set_xinput_devices_left_handed (left_handed);
+                return;
         }
 #endif
